pulsar/internal/connection.go

```diff
@@ -656,20 +662,14 @@ func (c *connection) Close() {
 
 	c.log.Info("Connection closed")
 	c.state = connectionClosed
-	if c.cnx != nil {
-		c.cnx.Close()
-	}
+	c.TriggerClose()
```

Comment:
If a client calls this won't it trigger another call to `Close()` if the run loop is still going. This would cause all the listeners to be trigger again. Are there potential issues with that?

Reply:
This method is wrapped with a mutex and a `c.state == connectionClosed` check.
I think another call to `Close()` will have no effect.